### PR TITLE
Rpc transaction routing

### PR DESCRIPTION
**Pull request by bowenwang1996**
Support transaction forwarding through `send_tx_commit` as well as transaction status querying routing. We should consider removing `TxDetails` since I don't see a use case for it @frol.
TODO: update near-shell to use the new API

### CODE REVIEW

File: chain/jsonrpc/src/lib.rs [hunk 1]
```diff
@@ -163,24 +179,43 @@ impl JsonRpcHandler {
...
+        let signer_account_id = tx.transaction.signer_id.clone();
         let result = self
             .client_addr
             .send(NetworkClientMessages::Transaction(tx))
             .map_err(|err| RpcError::server_error(Some(err.to_string())))
             .compat()
             .await?;
         match result {
-            NetworkClientResponses::ValidTx => {
+            NetworkClientResponses::ValidTx | NetworkClientResponses::NoResponse => {
```
Comment (SkidanovAlex) [inline chain/jsonrpc/src/lib.rs]: `NoResponse` should not mean "needs routing", we should add `NeedsRouting` response.

File: chain/jsonrpc/src/lib.rs [hunk 2]
```diff
@@ -163,24 +179,43 @@ impl JsonRpcHandler {
...
+                            let final_tx = self
+                                .client_addr
+                                .send(NetworkClientMessages::TxStatus { tx_hash, signer_account_id: signer_account_id.clone() })
+                                .compat()
+                                .await;
+                            if let Ok(NetworkClientResponses::TxStatus(ref tx_result)) = final_tx {
+                                match tx_result.status {
+                                    FinalExecutionStatus::Started
+                                    | FinalExecutionStatus::NotStarted => {}
+                                    _ => {
```
Comment (SkidanovAlex) [inline chain/jsonrpc/src/lib.rs]: List the outcomes, don't use `_`

**Comment by frol** [comment]
@bowenwang1996 I have just learned that this has been a breaking change for the `tx` (`txStatus`) API. I would like to learn more about the requirement of having `account_id` passing to query transaction information as it seems that even if I pass an empty string, it works fine.

**Comment by frol** [comment]
It casts Explorer to either require account ID (together with transaction hash) passed via URL or rely on the syncing backend to discover the transaction in blocks/chunks. The later won't allow exploring the transactions that are not in chunks yet (though, I am not sure if we need this). Should I add account ID to the transaction details URL on Explorer?

**Comment by bowenwang1996** [comment]
hmm it seems to me that transactions that are not in chunks yet should not be displayed because that means the transaction is not confirmed.

**Pull request merged**